### PR TITLE
Accountsdb: support config in Json5

### PR DESCRIPTION
**Pull request by fanatid**
#### Problem

Accountsdb uses JSON as a format for config files but JSON does not support comments.

#### Summary of Changes

In addition to JSON support YAML.

### CODE REVIEW

**Comment by mvines** [comment]
Alternatively we could just move from JSON to JSON5: https://crates.io/crates/json5

**Comment by fanatid** [comment]
I did not know about JSON5 :open_mouth: thanks!

But editors and other tools probably will not parse JSON files correctly :disappointed:

**Comment by mvines** [comment]
JSON5 has been around for a while, just need to use a .json5 extension to avoid editors from getting confused

**Comment by t-nelson** [comment]
I have no opinion on _what_ is switched to, only that there is a full switch, not multiple supported config file formats.

**Comment by fanatid** [comment]
> I have no opinion on what is switched to, only that there is a full switch, not multiple supported config file formats.

ok, should I change JSON to JSON5?

**Comment by mvines** [comment]
JSON5 is my vote as it's backwards compatible with the existing JSON format, so no migration for existing users **and** we get the comment support that this PR seeks to add

**Comment by lijunwangs** [comment]
> JSON5 is my vote as it's backwards compatible with the existing JSON format, so no migration for existing users **and** we get the comment support that this PR seeks to add

I second that -- no need to have another file format just for configuration purpose to keep it simple.

File: accountsdb-plugin-manager/Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ documentation = "https://docs.rs/solana-validator"
 [dependencies]
 bs58 = "0.4.0"
 crossbeam-channel = "0.5"
+json5 = "0.4.1"
 libloading = "0.7.2"
 log = "0.4.11"
 serde = "1.0.133"
```
Comment (lijunwangs) [inline accountsdb-plugin-manager/Cargo.toml:18]: Have you seen if these serde dependencies can be removed?

File: accountsdb-plugin-manager/src/accountsdb_plugin_service.rs [hunk 2]
```diff
@@ -156,12 +155,12 @@ impl AccountsDbPluginService {
...
         }
 
-        let result: serde_json::Value = match serde_json::from_str(&contents) {
+        let result: serde_json::Value = match json5::from_str(&contents) {
             Ok(value) => value,
             Err(err) => {
                 return Err(AccountsdbPluginServiceError::InvalidConfigFileFormat(
                     format!(
-                        "The config file {:?} is not in a valid Json format, error: {:?}",
+                        "The config file {:?} is not in a valid Json5 format, error: {:?}",
```
Comment (lijunwangs) [inline accountsdb-plugin-manager/src/accountsdb_plugin_service.rs:163]: Can you update the doc in https://github.com/solana-labs/solana/blob/master/docs/src/developing/plugins/accountsdb_plugin.md to reflect it uses JSON5 format.

File: accountsdb-plugin-manager/Cargo.toml (hunk 1 above)
Comment (fanatid) [reply accountsdb-plugin-manager/Cargo.toml]: Indeed, `serde` is not required. Removed.

**Comment by lijunwangs** [comment]
@fanatid Can you please resolve the merge conflict?

**Comment by lijunwangs** [comment]
> @fanatid Can you please resolve the merge conflict?

Anyway - I tried the resolving the conflict and let's see how the test goes.

**Review by lijunwangs — APPROVED** [review]
Looks good

**Pull request merged**